library/data/external/gsuite.py

```diff
@@ -139,41 +145,56 @@ def meet_spaces_client(self) -> meet_v2.SpacesServiceAsyncClient:
             self._spaces_client = meet_v2.SpacesServiceAsyncClient(credentials=credentials)
         return self._spaces_client
 
-    def __init__(self, email, creds):
-        self.email = email
+    def __init__(self, user: User, creds: str):
+        self.user = user
+        self.email = user.email
         self.creds = creds
+        self.is_local = os.getenv("IS_LOCAL") in ["True", "true", "1"]
+
+    def get_existing_redentials(self) -> Credentials:
+        creds: Credentials = None
+        if self.is_local:
+            if os.path.exists(self.TOKEN_FILE):
+                creds = Credentials.from_authorized_user_file(self.TOKEN_FILE, self.SCOPES)
+        else:
+            stored_creds: OAuthCreds = AuthManager().read_remote_credentials(self.user, DataSources.GOOGLE)
+            creds: Credentials = stored_creds.to_credentials() if stored_creds else None
+
+        return creds
+
 
     def __authenticate(self):
-        creds = None
-        # the file token.json stores the user's access and refresh tokens, and is
-        # created automatically when the authorization flow completes for the first time
-        if os.path.exists(self.TOKEN_FILE):
-            creds = Credentials.from_authorized_user_file(self.TOKEN_FILE, self.SCOPES)
+        creds: Credentials = self.get_existing_redentials()
 
         # if there are no (valid) credentials availablle, let the user log in.
         if not creds or not creds.valid:
             if creds and creds.expired and creds.refresh_token:
                 creds.refresh(Request())
-            else:
+            elif self.is_local:
```

Comment:
Is this to differentiate if the auth is happening from a local/prod environment?

Reply:
Yep. On local, we wtill need the browser auth to work on the api